Runtime error when specifying root directory on Windows

# Version
* Edition: Windows 11 Home 22H2
* OS: 22621.1555

# Detail
An error occurs if the root directory is specified.
```
>go-ls.exe  C:\    
The process cannot access the file because it is being used by another process.
C:\ :
Total files:  0
```

```
go-ls.exe -a ./ C:\
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
main.GetOwnerGroup({0xc000018dc0?, 0xf?}, {0x20?, 0xc000018cb8?})
        ./utils_windows.go:188 +0x5dc
main.GatherFileInfo({0xc000130200, 0x11, 0x0?})
        ./gather.go:24 +0x147
main.EnumFilePath({0xc0000240a0, 0x2, 0xc000018375?}, 0x1)
        ./main.go:50 +0x14a
main.main()
        ./main.go:69 +0x128
```

--- a/utils_windows.go
+++ b/utils_windows.go
@@ -178,6 +178,11 @@ func GetOwnerGroup(filePath string, fileinfo fs.FileInfo) (n string, dn string) 
 		&domainLen,
 		&sidUse,
 	)
+	// If the function fails, it returns zero.
+	if nameLen == 0 {
+		n, dn = "<UNKNOWN>", "<UNKNOWN>"
+		return
+	}
 
 	// Allocate memory for size returned by previous call
 	name := make([]uint16, nameLen)
@@ -212,12 +217,8 @@ func RemoveHiddenFile(files []string) []string {
 			break
 		}
 		attributes, err := syscall.GetFileAttributes(pointer)
-		if err != nil {
-			fmt.Println(err)
-			break
-		}
-		if attributes&syscall.FILE_ATTRIBUTE_HIDDEN != 0 {
-			// Hidden files
+		if err != nil || attributes&syscall.FILE_ATTRIBUTE_HIDDEN != 0 {
+			// GetFileAttributes failed or Hidden files
 			continue
 		}
 		tmp = append(tmp, file)

